server/models/orderModel.js

```diff
@@ -0,0 +1,55 @@
+const mongoose = require("mongoose");
+
+const orderSchema = mongoose.Schema(
+  {
+    user: {
+      type: mongoose.Schema.Types.ObjectId,
+      required: true,
+      ref: "User",
+    },
+    orderItems: [
+      {
+        name: { type: String, required: true },
+        quantity: { type: Number, required: true },
+        image: { type: String, required: true },
+        price: { type: Number, required: true },
+        id: {
+          type: mongoose.Schema.Types.ObjectId,
+          required: true,
+          ref: "Recipe",

```

Comment:
@Vaibhav-Kambli I really think we need a chefId somewhere, otherwise fetching the orders of a chef won't be possible. I think since each order is assign to a chef, you can add that in your model. In frontend, you can get that information from cart and then pass it on along your body. 
if we decide to rely only on recipeId. we need to fetch all data from this collection populate the recipe field with its info and only from there we could have the info for the chef.which definitely is not a good approach.

Reply:
Yes, I will add the chef id based on the state value coming from your context.